src/Store/ExtensionsReducer.re

```diff
@@ -0,0 +1,13 @@
+/*
+ * ExtensionsReducer.re
+ */
+open Oni_Model;
+
+let reduce = (action: Actions.t, v: Extensions.t) => {
+  switch (action) {
+  | Extension(Activated(id)) => Extensions.activateExtension(id, v)
```

Comment:
The cause and effect is very unclear here. Either `Extensions(Activated(id))` does not actually mean an extension has been activated, but rather a request for activation, or `Extensions.activateExtension` doesn't actually activate the extension but just marks it as activated. At least one of these must be wrong.

Reply:
Good point - `Extensions.activeExtension` sounds like it is actually doing the activation - but in actuality, it's just recording that the activation happened.

I changed the naming of both these methods to `notifyActivated` and `notifyDiscovered`, but let me know if there was some other nomenclature you had in mind.

Reply:
Hmm, `notify` seems to me imply an active form of messaging. Perhaps `setActivated`, `markActivated` or `addToActivated` depending on the level of abstraction here?

Reply:
Thanks @glennsl ! Sounds good - I used `markActivated` instead of `notifyActivated`, and `add` instead of `notifyDiscovered`